api.go

```diff
@@ -158,12 +163,23 @@ func (api *CacophonyAPI) UploadThermalRaw(info *cptvInfo, r io.Reader) error {
 	req.Header.Set("Content-Type", w.FormDataContentType())
 	req.Header.Set("Authorization", api.token)
 
-	client := new(http.Client)
+	client := &http.Client{
+		Timeout: timeout,
+	}
 	resp, err := client.Do(req)
+	defer resp.Body.Close()
```

Comment:
The `defer` needs to be *after* the `if err...` check as `resp` will likely be `nil` if there's an error.